.github/workflows/coverity.yml

```diff
@@ -0,0 +1,41 @@
+name: Coverity
+
+on:
+  pull_request:
+    branches:
+      - "*"
+  push:
+    branches:
+      - main
+
+jobs:
+  scan:
+    name: "coverity scan"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Run coverify scan script
+        env:
+          COVERITY_SCAN_PROJECT_NAME: ${{ secrets.COVERITY_SCAN_PROJECT_NAME }}
+          COVERITY_SCAN_TOKEN: ${{ secrets.COVERITY_SCAN_TOKEN }}
+          COVERITY_EMAIL: ${{ secrets.COVERITY_EMAIL }}
+          GITHUB_REF: ${{ secrets.GITHUB_REF }}
+          GITHUB_SHA: ${{ secrets.GITHUB_SHA }}
```

Comment:
There are no variables GITHUB_REF and GITHUB_SHA in secrets. These are defined as ENV vars available to all runners out of the box, and also available in the `github.` context as `github.ref` and `github.sha`, see example just above this section https://docs.github.com/en/actions/reference/environment-variables#naming-conventions-for-environment-variables.

The current definition causes these 2 variables to be empty which can be seen in the logs and may also be the reason why Coverity gets confused b/c it doesn't know which commit sha is being analyzed.

Reply:
Hmm, I am not sure about that, please see upload instructions in https://scan.coverity.com/projects/osbuild-image-builder/builds/new,

Here is the example to upload scan result in instructions:

curl --form token=6nUoUYf9njsqg0mOwy4pBQ \
  --form email=yih@redhat.com \
  --form file=@tarball/file/location \
  --form version="Version" \
  --form description="Description" \
  https://scan.coverity.com/builds?project=osbuild%2Fimage-builder

I am not sure that coverity server can get commit information from description, because in above example, there is no GITHUB_REF and GITHUB_SHA in description, these two variables are not required.
It's just that I used these two variables as description so we know more build information of the analysis result, although it does not work :(

Reply:
Please change the tokens b/c they are now publicly exposed and don't post secrets in plain/text on GitHub!

Reply:
The reason b/c your description and version are empty is because you are overriding the existing variables with empty string b/c your `env` section in the yaml file is incorrect. Do you understand why that happens ?

Reply:
Yes, I know what you mean. I understand that the env variables GITHUB_REF and GITHUB_SHA I used in workflow is empty, because my yaml file is incorrect about this:
GITHUB_REF: ${{ secrets.GITHUB_REF }}
GITHUB_SHA: ${{ secrets.GITHUB_SHA }}

To fix this, instead of reading env secrets, I can use github.ref and github.sha directly in workflow like this:

          curl https://scan.coverity.com/builds?project="$COVERITY_SCAN_PROJECT_NAME" \
            --form token="$COVERITY_SCAN_TOKEN" \
            --form email="$COVERITY_EMAIL" \
            --form file=@cov-int.tar.gz \
            --form version="$(git rev-parse HEAD)" \
            --form description="$GITHUB_REF  / $GITHUB_SHA"

Is that ok?

Reply:
Or I can remove these two variables and use a simple 'description', like 'osbuild-composer coverity scan', because these two variables are not required.

Reply:
GITHUB_REF and GITHUB_SHA are already available to all runners, you don't have to redefine them. See the github documentation.